data-prepper-api/src/test/java/org/opensearch/dataprepper/typeconverter/BigDecimalConverterTests.java

```diff
@@ -0,0 +1,101 @@
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.Collections;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class BigDecimalConverterTests {
+    @Test
+    void testStringToBigDecimalConversion() {
+        BigDecimalConverter converter = new BigDecimalConverter();
+        final String stringConstant = "12345678912.12345";
+        assertThat(converter.convert(stringConstant), equalTo(new BigDecimal(stringConstant)));
+    }
+
+    @Test
+    void testIntegerToBigDecimalConversion() {
+        BigDecimalConverter converter = new BigDecimalConverter();
+        final int intConstant = 12345;
+        assertThat(converter.convert(intConstant), equalTo(BigDecimal.valueOf(intConstant)));
+    }
+
+    @Test
+    void testLongToBigDecimalConversion() {
+        BigDecimalConverter converter = new BigDecimalConverter();
+        final long longConstant = 123456789012L;
+        assertThat(converter.convert(longConstant).longValue(), equalTo(longConstant));
+    }
+
+    @Test
+    void testBooleanToBigDecimalConversion() {
+        BigDecimalConverter converter = new BigDecimalConverter();
+        final Boolean boolFalseConstant = false;
+        assertThat(converter.convert(boolFalseConstant), equalTo(BigDecimal.valueOf(0)));
+        final Boolean boolTrueConstant = true;
+        assertThat(converter.convert(boolTrueConstant), equalTo(BigDecimal.valueOf(1)));
+    }
+
+    @Test
+    void testFloatToBigDecimalConversion() {
+        BigDecimalConverter converter = new BigDecimalConverter();
+        final float fval = 12345.6789f;
+        assertThat(converter.convert(fval).floatValue(), equalTo(fval));
+    }
+
+    @Test
+    void testBigDecimalToBigDecimalConversion() {
+        BigDecimalConverter converter = new BigDecimalConverter();
+        BigDecimal bigDecimal = new BigDecimal("12345.6789");
+        assertThat(converter.convert(bigDecimal), equalTo(bigDecimal));
+    }
+
+    @ParameterizedTest
+    @MethodSource("decimalToBigDecimalValueProvider")
+    void testDoubleToBigDecimalConversion(BigDecimal expectedBigDecimal, double actualValue, int scale) {
+        BigDecimalConverter converter = new BigDecimalConverter();
+        if(scale!=0) {
+            expectedBigDecimal = expectedBigDecimal.setScale(scale, RoundingMode.HALF_EVEN);
+        }
+        assertThat(converter.convert(actualValue, scale), equalTo(expectedBigDecimal));
+    }
+
+    private static Stream<Arguments> decimalToBigDecimalValueProvider() {
+        return Stream.of(
+            Arguments.of(new BigDecimal ("0.0"), 0, 1),
```

Comment:
Can we add another test case that covers the conversion described in the issue here (https://github.com/opensearch-project/data-prepper/issues/3840) where we can convert `1.70206220242E+12` to `702062202420`

Reply:
Added additional scenario for this case